full-heal.lua

```diff
@@ -148,16 +161,68 @@ if unit then
         status.skin_damage = false
         status.motor_nerve_severed = false
         status.sensory_nerve_severed = false
+        status.spilled_guts = false
+        status.severed_or_jammed = false
     end
 
-    for i, temp in ipairs(unit.status2.body_part_temperature) do
-        local bp = unit.body.body_plan.body_parts[i]
-        temp.whole = math.floor((bp.min_temp + bp.max_temp) / 2)
+    for i = #unit.status2.body_part_temperature-1,0,-1 do
+      unit.status2.body_part_temperature:erase(i) -- attempting to rewrite temperature was causing body parts to melt for some reason; forcing repopulation in this manner appears to be safer
     end
 
-    if unit.job.current_job and unit.job.current_job.job_type == df.job_type.Rest then
-        --print("Wake from rest -> clean self...")
-        unit.job.current_job.job_type = df.job_type.CleanSelf
+    for i = 0,#unit.enemy.body_part_8a8-1,1 do
+      unit.enemy.body_part_8a8[i] = 1 -- not sure what this does, but values appear to change following injuries
```

Comment:
What exactly are following injuries? Injuries related to this specific body part?

Reply:
Affirmative for 'injuries related to this specific body part', but I really don't know anything beyond that. To give a tangible example: severing a body part off changes the corresponding value (indexed by body part id) in body_part_878 to 0. Same for body_part_8a8 and body_part_888.